.github/workflows/ci_code_quality.yml

```diff
@@ -42,14 +42,14 @@ jobs:
       - name: Check out the repository
         uses: actions/checkout@v4
 
-      - name: Set up Python
-        uses: actions/setup-python@v5
+      - name: "Set up Python & Hatch - 3.11"
+        uses: ./.github/actions/setup-python-hatch
         with:
-          python-version: '3.11'
+          python-version: "3.11"
 
-      - name: Install Hatch
+      - name: Install pre-commit
```

Comment:
Why would we install pre-commit in CI? I'm probably answering my own question, but is it to catch the eof and trailing whitespace issues?

Reply:
You are indeed answering your own question.